Mastodon/Scene/Notification/Notification Filtering/Requests/NotificationRequestsTableViewController.swift

```diff
@@ -0,0 +1,226 @@
+                var snapshot = NSDiffableDataSourceSnapshot<NotificationRequestsSection, NotificationRequestItem>()
+                snapshot.appendSections([.main])
+                snapshot.appendItems(self.viewModel.requests.compactMap { NotificationRequestItem.item($0) } )
+
+                self.dataSource?.apply(snapshot)
+            }
+        } else {
+            await MainActor.run { [weak self] in
+                _ = self?.navigationController?.popViewController(animated: true)
+            }
+        }
+
+    }
+
+    func rejectNotificationRequest(_ cell: NotificationRequestTableViewCell, notificationRequest: MastodonSDK.Mastodon.Entity.NotificationRequest) {
+
+        cell.rejectNotificationRequestActivityIndicatorView.isHidden = false
+        cell.rejectNotificationRequestActivityIndicatorView.startAnimating()
+        cell.rejectNotificationRequestButton.tintColor = .clear
+        cell.rejectNotificationRequestButton.setTitleColor(.clear, for: .normal)
+        cell.rejectNotificationRequestButton.isUserInteractionEnabled = false
+        cell.acceptNotificationRequestButton.isUserInteractionEnabled = false
+
+        Task { [weak self] in
+            guard let self else { return }
+            do {
+                try await rejectNotificationRequest(notificationRequest)
+            } catch {
+                cell.rejectNotificationRequestActivityIndicatorView.stopAnimating()
+                cell.rejectNotificationRequestButton.tintColor = .black
+                cell.rejectNotificationRequestButton.setTitleColor(.black, for: .normal)
+                cell.rejectNotificationRequestButton.isUserInteractionEnabled = true
+                cell.acceptNotificationRequestButton.isUserInteractionEnabled = true
+            }
+        }
+    }
+
+    private func rejectNotificationRequest(_ notificationRequest: MastodonSDK.Mastodon.Entity.NotificationRequest) async throws {
+        _ = try await context.apiService.rejectNotificationRequests(authenticationBox: authContext.mastodonAuthenticationBox,
+                                                                    id: notificationRequest.id)
+
+        let requests = try await context.apiService.notificationRequests(authenticationBox: authContext.mastodonAuthenticationBox).value
+
+        NotificationCenter.default.post(name: .notificationFilteringChanged, object: nil)
+
+        if requests.count > 0 {
+            await MainActor.run { [weak self] in
+                guard let self else { return }
+                self.viewModel.requests = requests
+                var snapshot = NSDiffableDataSourceSnapshot<NotificationRequestsSection, NotificationRequestItem>()
+                snapshot.appendSections([.main])
+                snapshot.appendItems(self.viewModel.requests.compactMap { NotificationRequestItem.item($0) } )
+
+                self.dataSource?.apply(snapshot)
+            }
+        } else {
+            await MainActor.run { [weak self] in
```

Comment:
Just the same idea as above, but if you think it's more reasonable to keep the two individual `MainActor.run` calls here then please ignore my comment(s).